pulp_container/app/viewsets.py

```diff
@@ -985,26 +946,21 @@ class ContainerPushRepositoryVersionViewSet(
     """
 
     parent_viewset = ContainerPushRepositoryViewSet
-    queryset_filtering_required_repo_permission = "container.view_containerpushrepository"
+    permission_classes = (access_policy.PushRepositoryVersionAccessPolicy,)
 
     DEFAULT_ACCESS_POLICY = {
         "statements": [
             {
-                "action": ["list"],
-                "principal": "authenticated",
-                "effect": "allow",
-            },
-            {
-                "action": ["retrieve"],
+                "action": ["list", "retrieve"],
                 "principal": "authenticated",
                 "effect": "allow",
-                "condition": "has_repo_attr_model_or_obj_perms:container.view_containerpushrepository",  # noqa
+                "condition": "has_distribution_perms:container.view_containerdistribution",
```

Comment:
Do we need to check namespace perms as well? 
It does not seem that the distribution perm check will look at the namespace perms.
I'm not sure though if in the current workflows it is possible to have a model or obj level permission for the namespace but not for the distribution of a push repo.

Reply:
You are probably right.